content/tutorials/reactor-event-cloudflare.textile

```diff
@@ -0,0 +1,256 @@
+
+
+<a href="/images/tutorials/reactor/reactor-event-cloudflare-game-screenshot.png" target="_blank">
+  <img src="/images/tutorials/reactor/reactor-event-cloudflare-game-screenshot.png" style="width: 100%" alt="screenshot of the game we will be making">
+</a>
+
+
+h2. Ably Reactor Prerequisites
+
+We're going to need an Ably API key for use by your browser application, and for the Cloudflare Worker, to allow it to post a response back to Ably. For our purposes, only one API key is required. If you extend the tutorial with your own functionality, you may wish to create a separate API key for the Cloudflare Worker if you need it to have a different set of permissions than those assigned for use by the browser app.
+
+First, log into your Ably dashboard, find the app you're going to be using for this tutorial (the "Sandbox" app is probably fine), then click "Manage App".
+
+<a href="/images/tutorials/reactor/reactor-event-ably-apikey-1.png" target="_blank">
+  <img src="/images/tutorials/reactor/reactor-event-ably-apikey-1.png" style="width: 100%" alt="Manage your Ably app">
+</a>
+
+Select the "API Keys" tab, and click "Create a new API key":
+
+<a href="/images/tutorials/reactor/reactor-event-ably-apikey-2.png" target="_blank">
+  <img src="/images/tutorials/reactor/reactor-event-ably-apikey-2.png" style="width: 100%" alt="Create a new API key">
+</a>
+
+# Enter a name for your API key
+# Select only the "publish" and "subscribe" options
+# Click the "Create key" button
+
+Scroll down if necessary, and copy and paste your new API key to a text file for later use in your application.
+
+h2. Building the Game
+
+To get you up and running quickly, we've implemented this 2D game and made it available via this "tutorial's repository":https://github.com/ably/tutorials/tree/reactor-event-cloudflare-javascript. You can clone this repository and install its dependencies as per its readme.
+
+h2. Step 2: Creating your function
+
+Cloudflare workers is a Function-as-a-Service (FaaS) platform operated by Cloudflare that lets you run pieces of code when they are triggered. They're comparable to AWS Lambda, or Azure Functions.
+
+Cloudflare have an npm tool called "Wrangler":https://developers.cloudflare.com/workers/quickstart/ that they use to create their Workers. You can install this using:
+
+```[bash]
+npm install -g @cloudflare/wrangler
+```
+
+and then you can create new Workers by using the command `wrangler generate <my-worker-name>`.
+
+If you want to learn more about Wrangler and Cloudflare Workers, you can "read their documentation":https://developers.cloudflare.com/workers/ on the Cloudflare developer site.
+
+For this example, we've already done this for you and you'll find a Worker in the *./cloudflare-worker/* directory in your cloned repository. It looks a little like this:
+
+```[js]
```

Comment:
This shoudl be `javascript`, see https://docs.ably.io/client-lib-development-guide/documentation-formatting-guide/#definition-lists